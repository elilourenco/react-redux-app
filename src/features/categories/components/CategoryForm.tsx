import { Box, Button, FormControl, FormControlLabel, 
FormGroup, Grid, Paper, Switch, TextField} from "@mui/material"
import { Category, } from "../categorySlice";
import { Link as RouterLink} from "react-router-dom";


type Props = {
    category:Category;
    isdisabled?:boolean;
    isLoading?:boolean;
    handleSubmit:(e:React.FormEvent<HTMLFormElement>) => void;
    handleChange:(e:React.ChangeEvent<HTMLInputElement>) => void;
    handleToggle:(e:React.ChangeEvent<HTMLInputElement>) => void;

}


export  function CategoryForm ({
    category,
    isdisabled=false,
    isLoading=false,
    handleSubmit,
    handleChange,
    handleToggle,

}:Props) {
 return(
    <Paper>
        <Box p={2}>  
            <form onSubmit={handleSubmit} >
                <Grid container spacing={3}>

                    <Grid  size={{xs:12}}>
                        <FormControl fullWidth>
                            <TextField 
                                required
                                    name="first_Name"
                                    label="first_Name"
                                    value={category.first_Name}
                                    disabled={isdisabled}
                                    onChange={handleChange}
                                />
                            </FormControl>

                        </Grid> 

                        <Grid size={{xs:12}} >
                            <FormControl fullWidth>
                                <TextField
                                    required
                                    name="last_Name"
                                    label="last_Name"
                                    value={category.last_Name}
                                    disabled={isdisabled}
                                    onChange={handleChange}
                                />
                            </FormControl>

                        </Grid>

                                                <Grid size={{xs:12}} >
                            <FormControl fullWidth>
                                <TextField
                                    required
                                    name="email"
                                    label="email"
                                    value={category.email}
                                    disabled={isdisabled}
                                    onChange={handleChange}
                                />
                            </FormControl>

                        </Grid>


                        

                        {/*<Grid size={{xs:12}} >
                            <FormGroup>
                                <FormControlLabel
                                    control={
                                        <Switch
                                            name="is_active"
                                            color="secondary"
                                            checked ={category.is_active}
                                            onChange={handleChange}
                                            disabled={isdisabled}
                                            aria-label="Toggle switch"
                                        />
                                    }
                             
                                    label="Active"
                                />                      
                            
                            </FormGroup>

                        </Grid> */}

                        <Grid size={{xs:12}} >
                            <Box display="flex" gap={2}>
                                <Button  
                                    variant="contained" 
                                    component={RouterLink} 
                                    to="/categories">
                                   Back
                                 
                                </Button>

                                <Button 
                                type="submit"
                                variant="contained"
                                color="secondary"
                                disabled={isdisabled || isLoading}
                                >
                                
                                {isLoading ?  "Saving..." : "Save"}
                                    
                                </Button>
                            </Box>
                        </Grid>
                        


                    </Grid>
                </form>
 
            </Box>
        </Paper>
)}