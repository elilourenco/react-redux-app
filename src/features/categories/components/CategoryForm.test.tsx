import { render } from "@testing-library/react";
import { CategoryForm } from "./CategoryForm";
import { BrowserRouter } from "react-router-dom";
import { it } from "node:test";



type  category ={
    id:number;
  first_Name: string;
  last_Name:string;
  email: string;
}

const  Props = {
    category:{} as  category,
    isdisabled:false,
    isLoading:false,
    handleSubmit:() => {},
    handleChange:() => {},
    handleToggle:() => {},

}


describe("CategoryForm",()=>{
    it("Should  render correctly",()=>{
        const {asFragment} = render(<CategoryForm {...Props} />,{
            wrapper: BrowserRouter,
        });
         expect( asFragment()).toMatchSnapshot()
    })

    it("Should",()=>{

        const {asFragment} = render(<CategoryForm {...Props} isLoading={null as any} isdisabled={null as any} />,{
            wrapper: BrowserRouter,
        });
         expect( asFragment()).toMatchSnapshot()
        
    })
})