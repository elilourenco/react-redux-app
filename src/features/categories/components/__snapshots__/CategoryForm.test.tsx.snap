// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryForm Should  render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1gtchvp-MuiPaper-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiBox-root css-19midj6"
    >
      <form>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-3 css-1u8xl8z-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="«r0»"
                  id="«r0»-label"
                >
                  first_Name
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                    id="«r0»"
                    name="first_Name"
                    required=""
                    type="text"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1n64csd-MuiNotchedOutlined-root"
                    >
                      <span>
                        first_Name *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="«r1»"
                  id="«r1»-label"
                >
                  last_Name
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                    id="«r1»"
                    name="last_Name"
                    required=""
                    type="text"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1n64csd-MuiNotchedOutlined-root"
                    >
                      <span>
                        last_Name *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="«r2»"
                  id="«r2»-label"
                >
                  email
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                    id="«r2»"
                    name="email"
                    required=""
                    type="text"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1n64csd-MuiNotchedOutlined-root"
                    >
                      <span>
                        email *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-10egq61"
            >
              <a
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
                href="/categories"
                tabindex="0"
              >
                Back
              </a>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-1mns5ui-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Save
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
